compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -3182,37 +3179,29 @@ impl<'tcx> TypeErrCtxtExt<'tcx> for TypeErrCtxt<'_, 'tcx> {
             let mut type_diffs = vec![];
 
             if let ObligationCauseCode::ExprBindingObligation(def_id, _, _, idx) = parent_code.deref()
-                && let predicates = self.tcx.predicates_of(def_id).instantiate_identity(self.tcx)
```

Comment:
Instantiating identity substitutions will cause us to collect unnecesary type errors that wouldn't show up after substitution -- see the UI test that changed.